modules/tinymce/src/core/main/ts/util/Quirks.ts

```diff
@@ -327,6 +329,160 @@ const Quirks = (editor: Editor): Quirks => {
+  // it with the original content saved on keydown which does not have font-family
+  const removeExtraFontFamilyOnKeyup = (editor: Editor, specialDelete: Cell<boolean>, content: Cell<Optional<string>>) => {
+    editor.on('keyup', (e) => {
+      if (isDefaultPrevented(e) || e.key !== 'Backspace' && e.key !== 'Delete' || !specialDelete.get()) {
+        return;
+      }
+
+      const rng = selection.getRng();
+      const container = rng.startContainer;
+      const root = dom.getRoot();
+      const parent = Optional.from(Traverse.parents(SugarElement.fromDom(container)).find((node) => node.dom.nodeName.toLowerCase() === 'li'));
+
+      let outsideContainer = container;
+      while (
+        outsideContainer.parentNode &&
+        outsideContainer.parentNode.firstChild === outsideContainer &&
+        outsideContainer.parentNode !== root &&
+        outsideContainer.parentNode.nodeName.toLowerCase() !== 'li'
+      ) {
+        outsideContainer = outsideContainer.parentNode;
+      }
+      let outsideOffset = Optional.none<number>();
+      parent.each((parent) => {
+        parent.dom.childNodes.forEach((node, key) => {
+          if (node === outsideContainer) {
+            outsideOffset = Optional.some(key);
+            return;
+          }
+        });
+        outsideOffset.each((offset) => {
+          selection.getSel()?.setBaseAndExtent(parent.dom, offset + 1, parent.dom, parent.dom.childNodes.length);
+          editor.execCommand('Delete');
+          selection.setCursorLocation(parent.dom, offset + 1);
+          content.get().each((content) => editor.insertContent(content));
+          selection.setCursorLocation(parent.dom, offset + 1);
+        });
+
+      });
+
+      specialDelete.set(false);
+    });
+  };
+
+  /**
+   * Removes font-family style added when pressing backspace when the cursor is just before an image
+   * and there is a list before the image. #TINY-10892
+   */
+  const removeExtraFontFamilyOnBackspace = () => {
+    const specialDelete = Cell(false);
+    const content = Cell<Optional<string>>(Optional.none());
+    editor.on('keydown', (e) => {
+      if (isDefaultPrevented(e) || e.key !== 'Backspace') {
+        return;
+      }
+
+      const rng = selection.getRng();
+      const container = rng.startContainer;
+      let parent = container;
+      const offset = rng.startOffset;
+      const root = dom.getRoot();
+
+      if (!rng.collapsed || offset !== 0) {
+        return;
+      }
+
+      while (
+        parent.parentNode &&
+        parent.parentNode.firstChild === parent &&
+        parent.parentNode !== root
+      ) {
+        parent = parent.parentNode;
+      }
+      let hasImgNode = false;
+      parent.childNodes.forEach((node) => {
+        if (node.nodeName.toLowerCase() === 'img') {
+          hasImgNode = true;
+          return;
+        }
+      });
+      if (!hasImgNode || parent.previousSibling?.nodeName.toLowerCase() !== 'ol') {
```

Comment:
This doesn't work with `UL` should handle both cases. There is a ElementType.isList that takes sugar elements. Maybe if the code was all sugar then one could use that.

Reply:
Done!